agent/consul/leader_connect_ca.go

```diff
@@ -873,12 +873,13 @@ func (c *CAManager) UpdateConfiguration(args *structs.CARequest) (reterr error)
 				"You can try again with ForceWithoutCrossSigningSet but this may cause " +
 				"disruption - see documentation for more.")
 		}
-		if !canXSign && args.Config.ForceWithoutCrossSigning {
-			c.logger.Warn("current CA doesn't support cross signing but " +
-				"CA reconfiguration forced anyway with ForceWithoutCrossSigning")
+		if args.Config.ForceWithoutCrossSigning {
+			c.logger.Warn("ForceWithoutCrossSigning set, CA reconfiguration skipping cross-signing")
```

Comment:
Wouldn't it make more sense to move this into the `if canXSign && ...` block below so that you don't get a warning in the case that you are using a provider where `canXSign==false`?

Reply:
That block only happens when we *aren't* skipping cross signing though, so I think this is the right place for the log statement. This change is to ignore the value of `canXsign` when `ForceWithoutCrossSigning` is set

Reply:
ah ok